### PR TITLE
pkg/trace: Fix flaky tests due to metrics race with Timing package

### LINKED ISSUE #13934: [Flaky Test] `FuzzHandleTraces*` tests are very flaky
It seems this is due to the `timing` package spinning up a goroutine on `init` that accesses the `metrics` client. Many tests will overwrite the metrics client with a test version (thus creating a data race and failing the test)

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
fixes #13934

### Motivation
Data race with `timing` package means these tests would flake fail with the race detector. Namely the `timing` package would auto run the `Autoreport` function which would call `metrics.Count` and `metrics.Gauge` racing with tests which overwrite the stats client.

### Possible Drawbacks / Trade-offs
A better solution would probably be to not just auto start a goroutine in the `init` of the `timing` package. However since it's already there and semi-widely used I wanted to avoid introducing a larger and hard to test change when the goal here was to reduce flakiness.

### Describe how to test/QA your changes
These tests are pretty difficult to get to flake. I got lucky occasionally running:
`go test ./... -count=40 -tags otlp -race -p 4`
And after these changes haven't been able to reproduce the flake.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by knusbaum — APPROVED** [review]
LGTM. Thanks, @ajgajg1134

File: pkg/trace/api/listener_test.go [hunk 1]
```diff
@@ -116,6 +117,7 @@ func TestMockListener(t *testing.T) {
 
 func TestMeasuredListener(t *testing.T) {
 	assert := assert.New(t)
+	timing.Stop() // https://github.com/DataDog/datadog-agent/issues/13934
```
Comment (gbbr) [inline pkg/trace/api/listener_test.go:120]: Have you considered the situation where someone forgets to do this in the future and introduces another flaky test? 

Considering that this only needs to happen when replacing the metrics client, I'd suggest adding a method for that into the `testutil` package, e.g.
```go
// WithStatsClient replaces the global metrics.StatsClient with c. It also returns
// a function for restoring the original client.
func WithStatsClient(c metrics.StatsClient) func() {
    old := metrics.Client
    timing.Stop()  // https://github.com/DataDog/datadog-agent/issues/13934
	metrics.Client = c
    return func() { metrics.Client = old }
}
```
Then, you can use it in the tests:
```go
var stats testutil.TestStatsClient
defer testutil.WithStatsClient(&stats)()
// ...
```
Less boilerplate and less risk of forgetting to stop the timing reporter and introduce a flaky test.

**Review by katiehockman — APPROVED** [review]
LGTM (after @gbbr's suggestion is taken)

File: pkg/trace/testutil/span.go [hunk 2]
```diff
@@ -167,7 +167,7 @@ var metas = map[string][]string{
 	},
 }
 
-var metrics = []string{
+var spanMetrics = []string{
```
Comment (ajgajg1134) [inline pkg/trace/testutil/span.go:170]: I renamed this to avoid colliding with a `metrics` package import

**Review by katiehockman — APPROVED** [review]
It's a little weird that the teststatsd package doesn't actually have the `WithStatsClient` function, but the reasoning around the cyclic dependency make sense, so this looks better to me 👍

**Review by Julio-Guerra — APPROVED** [review]
I've got good news: the old appsec event intake is now totally disabled and it is now safe to remove it from the agent too.

I'll open a PR for you removing these things from the trace-agent since ASM relies a 100% on the regular APM intake now :-)

**Pull request merged**